src/lib.rs

```diff
@@ -128,6 +132,20 @@ pub fn collect_stars(config: Config) -> Result<(), Box<dyn error::Error>> {
     }
     println!("Collected {} stars", stars.len());
 
+    let path = Path::new("output.md");
+    let display = path.display();
+
+    let mut file = match File::create(&path) {
+        Err(why) => panic!("couldn't create {}: {}", 
+                            display, 
+                            why.description()),
+        Ok(file) => file,
+    };
+
+    for star in stars.iter(){ 
+        write!(file, "{}\x20\x20\r", star);
```

Comment:
`\x20` is indeed a space - in order to get the markdown on the next line rather than in the same line, it needs 2 spaces and a carriage return/newline. 

`\r` is a Rust token: https://doc.rust-lang.org/reference/tokens.html - but if it does fail using that on linux then `writeln!` may be better if it always makes sure there's a newline regardless of OS.

Reply:
> to get the markdown on the next line rather than in the same line, it needs 2 spaces and a carriage return/newline

Huh, a close reading of the [original Markdown spec](https://daringfireball.net/projects/markdown/syntax#p) does specify 2 spaces before a return.

I'm not actually aware of any other specs that specify this - [GitHub's Markdown spec](https://github.github.com/gfm/#characters-and-lines) says "A line is a sequence of zero or more characters other than newline (`U+000A`) or carriage return (`U+000D`), followed by a line ending or by the end of file" which doesn't mention spaces at all - and requiring 2 spaces will likely trigger trailing whitespace warnings in many editors.

> `\r` is a Rust token

Yup! Though `\r` is a [standard escape sequence](https://en.wikipedia.org/wiki/Carriage_return#Computers) so it's not Rust-specific. The problem is on Windows pressing the `ENTER` key generates the key sequence `\r\n` whereas on most other operating systems `ENTER` results in just `\n`. Microsoft only got around to [adding support for Unix/Linux line endings in Nodepad](https://blogs.msdn.microsoft.com/commandline/2018/05/08/extended-eol-in-notepad/) this year. You can use `write!` and add the linefeed yourself, but `writeln!` does that for you so it's up to you which one you'd like to use!